x/staking/client/cli/tx.go

```diff
@@ -337,51 +361,37 @@ $ %s tx staking cancel-unbond %s1gghjut3ccd8ay0zduzj64hwre2fxs9ldmqhffj 100stake
 	return cmd
 }
 
-func newBuildCreateValidatorMsg(clientCtx client.Context, txf tx.Factory, fs *flag.FlagSet) (tx.Factory, *types.MsgCreateValidator, error) {
-	fAmount, _ := fs.GetString(FlagAmount)
-	amount, err := sdk.ParseCoinNormalized(fAmount)
+func newBuildCreateValidatorMsg(clientCtx client.Context, txf tx.Factory, fs *flag.FlagSet, val validator) (tx.Factory, *types.MsgCreateValidator, error) {
+	amount, err := sdk.ParseCoinNormalized(val.Amount)
 	if err != nil {
 		return txf, nil, err
 	}
 
 	valAddr := clientCtx.GetFromAddress()
-	pkStr, err := fs.GetString(FlagPubKey)
-	if err != nil {
-		return txf, nil, err
-	}
 
 	var pk cryptotypes.PubKey
-	if err := clientCtx.Codec.UnmarshalInterfaceJSON([]byte(pkStr), &pk); err != nil {
+	if err := clientCtx.Codec.UnmarshalInterfaceJSON([]byte(val.PubKey), &pk); err != nil {
 		return txf, nil, err
 	}
 
-	moniker, _ := fs.GetString(FlagMoniker)
 	identity, _ := fs.GetString(FlagIdentity)
 	website, _ := fs.GetString(FlagWebsite)
```

Comment:
I am confused, are we now combining flags and a json?

Reply:
In the JSON, we are allowing only the required fields to create a validator as of now, so for creating the description it is necessary to combine flags(which are optional) and a json field.
Otherwise I could add these optional fields to validator struct and we can use only json fields to create this description which is completely fine IMO.

Reply:
Personally I think it is a bit weird to have to combine a JSON and flags. (Except from the from, which makes sense and we do that for proposals so it is still the name UX).

If you could indeed mark these fields as optional and in the JSON directly I think it makes more sense.